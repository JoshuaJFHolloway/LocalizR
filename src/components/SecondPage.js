import React, { Component } from 'react';
import SubHeader from './SubHeader';
import Picture from './Picture';



class SecondPage extends Component {
  
  constructor(){
    super();
    this.state = {
      spanish: null
    };
  };

  handlePictureClicked = () => {
    this.setState ({spanish: true});
  }

  render(){
    return (
      <div>
        <SubHeader
          subheader={"Choose your language!"}
        />
        <Picture 
        handlePictureClicked = {this.handlePictureClicked}
        />
      </div>
    );

  }
  
  
};

export default SecondPage;
