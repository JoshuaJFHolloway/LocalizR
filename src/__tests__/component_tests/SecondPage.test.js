import React from 'react';
import { shallow } from 'enzyme';
import SecondPage from '../../components/SecondPage';

describe('SecondPage', () => {
  let secondpage = shallow(<SecondPage />);

  describe('Rendering components', () => {

    it('renders correctly', () => {
      expect(secondpage).toMatchSnapshot();
    })

    it('renders Header component', () => {
      expect(secondpage.find('SubHeader').exists()).toBe(true);
    });

    it("renders Picture component", () => {
      expect(secondpage.find('Picture').exists()).toBe(true);
    });
  })

  describe('Initializes state successfully', () =>{
    it('initializes spanish as null', () => {
      expect(secondpage.state('spanish')).toEqual(null);
    });
  });

  describe('Updates state successfully', () => {
    it("Updates the state when clicks spanish Picture component", () => {
      secondpage.instance().handlePictureClicked();
      expect(secondpage.state('spanish')).toEqual(true);
    })
  })
});
