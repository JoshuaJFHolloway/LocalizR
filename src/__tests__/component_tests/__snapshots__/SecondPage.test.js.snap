// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondPage Rendering components renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SecondPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          Second Page has rendered, let the fun begin
</button>,
        <ChooseLanguage />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Second Page has rendered, let the fun begin",
        },
        "ref": null,
        "rendered": "Second Page has rendered, let the fun begin",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            Second Page has rendered, let the fun begin
</button>,
          <ChooseLanguage />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Second Page has rendered, let the fun begin",
          },
          "ref": null,
          "rendered": "Second Page has rendered, let the fun begin",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
